README.md

```diff
@@ -0,0 +1,61 @@
+# How to contribute a snippet
+
+1. Write the SQL to retrieve the data of interest.
+    * Try to put as much of the data wrangling logic into SQL as possible. This means there's less data wrangling code to write twice (e.g., once in [pandas](https://pandas.pydata.org/) for Python and again [dplyr](https://dplyr.tidyverse.org/) for R).
+    * Choose a good prefix for your `<my-query>.sql` filename. The file name helps users decide whether its useful to them. It will also become the name of the dataframe holding the query results (e.g. `<my-query>_df`).
+    * Put some comments in your query too. Look at the other `.sql` files for examples.
+    * Tip: Paste your SQL into the [BigQuery web UI](https://bigquery.cloud.google.com/) and click on the 'format' button to format it nicely.
+1. If your query has any parameter(s) other than `DATASET`, add default values for those parameters to both [`snippets_setup.R`](./snippets/snippets_setup.R) and [`snippets_setup.py`](./snippets/snippets_setup.py) so that the query will work as-is.
+1. [Optional] If your SQL is complex, consider writing a test case for it. Look at the other SQL tests for examples.
+1. If you want to include any visualizations for the data returned by your query, write a plot that assumes the existence of `<my-query>_df`.
+    * Write it first in [ggplot2](https://ggplot2.tidyverse.org/) or [plotnine](https://plotnine.readthedocs.io/en/stable/), depending on your preference, and iterate until you like the look of it.
+    * Once you have it the way you want it, port it to the other language.
+    * Name the files `<my-query>_<viz-description>.ggplot` and `<my-query>_<viz-description>.plotnine`
+
+Don't like these conventions? We can change them! This is just a starting point. Keep in mind we'll need to reflect those changes in the auto-generation script described in the next section.
+
+# Auto-generation of Jupyter 'Snippets Menu' configuration.
+
+For more detail, see the [Snippets Menu](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/snippets_menu/readme.html) Jupyter extension documentation.
+
+1. Generate the JSON configuration:
+    ```
+    git clone https://github.com/all-of-us/code-snippets.git
+    cd code-snippets/build
```

Comment:
nit: should there be a virtualenv + pip install step here?

Reply:
Added a comment about the pip command.